tests/test_plugin.py

```diff
@@ -9,56 +9,99 @@
-        [
-            b'--hello',
-        ]
+    # Clean then copy.
+    remove_all()
+    shutil.copytree(src_hello, dst_hello)
 
-    # Call DUNES.
+    # Call DUNES with '--help' flag and test for the expected result.
     completed_process = subprocess.run([DUNES_EXE, "--help"], check=True, stdout=subprocess.PIPE)
+    assert '--hello' in completed_process.stdout.decode()
 
-    # Test for expected values in the captured output.
-    for expect in expect_list:
-        assert expect in completed_process.stdout
+    # Call DUNES with '--hello' flag and test for the expected result.
+    completed_process = subprocess.run([DUNES_EXE, "--hello"], check=True, stdout=subprocess.PIPE)
+    assert 'Hello from DUNES' in completed_process.stdout.decode()
+
+    # Cleanup.
+    remove_all()
 
-    remove_plugin()
 
+def test_fail():
+    """Test that a duplicate working hello plugin."""
 
-def test_plugin_execution():
-    prepare_plugin()
+    # Clean then copy. Note that the fail plugin is a simple duplicate of hello.
+    remove_all()
+    shutil.copytree(src_hello, dst_hello)
+    shutil.copytree(src_hello, dst_fail)
 
-    expect_list = \
-        [
-            b'Hello from DUNES',
-        ]
+    # Call DUNES with '--help' flag and test for the expected result.
+    completed_process = subprocess.run([DUNES_EXE, "--help"], check=True, stdout=subprocess.PIPE)
+    assert '--hello' in completed_process.stdout.decode()
+    assert "Can't load module <" in completed_process.stdout.decode()
 
-    # Call DUNES.
+    # Call DUNES with '--hello' flag and test for the expected result.
     completed_process = subprocess.run([DUNES_EXE, "--hello"], check=True, stdout=subprocess.PIPE)
+    assert 'Hello from DUNES' in completed_process.stdout.decode()
+    assert "Can't load module <" in completed_process.stdout.decode()
+
+    # Cleanup.
+    remove_all()
+
+
+def test_missing_main():
+    """Test that a directory missing a main.py file."""
+
+    # Clean up then make the empty directory.
+    remove_all()
+    os.mkdir(dst_gotcha)
+
+    # Call DUNES with '--help' flag and test for the expected result.
+    completed_process = subprocess.run([DUNES_EXE, "--help"], check=True, stdout=subprocess.PIPE)
+    assert 'main.py not found in' in completed_process.stdout.decode()
+
+    remove_all()
+
+
+def test_gotcha():
+    """Test a bad plugin generates a notification."""
+
+    # Clean then copy.
+    remove_all()
+    shutil.copytree(src_gotcha, dst_gotcha)
+
+    # Call DUNES with '--help' flag and test for the expected result.
+    completed_process = subprocess.run([DUNES_EXE, "--help"], check=True, stdout=subprocess.PIPE)
+    assert 'Could not load' in completed_process.stdout.decode()
+
+    # Cleanup.
+    remove_all()
+
 
-    # Test for expected values in the captured output.
-    for expect in expect_list:
-        assert expect in completed_process.stdout
 
-    remove_plugin()
-    
\ No newline at end of file
+if __name__ == "__main__":
```

Comment:
test_xxx functions will be automatically picked by pytest, so this part if __name__ == "__main__" is not necessary

Reply:
The purpose of this code block is to allow a tester to easily run the test outside of pytest. Strictly speaking it is not necessary; however, I find the addition useful for debugging and other problem solving and so I prefer to leave it.